lib/base/utility.cpp

```diff
@@ -1265,6 +1266,25 @@ int Utility::CompareVersion(const String& v1, const String& v2)
 	return 0;
 }
 
+String Utility::ParseVersion(const String& version)
+{
+	bool ret;
+	boost::smatch match;
+
+	try {
+		boost::regex expr("[rv]?(\\d+.\\d+.\\d+).*");
```

Comment:
This regex is assuming Icinga-style version numbers. I'm OK with some Icinga-specific things, like the 'v' and 'r' prefixes. However, should we also support more generic version numbers like "3.9"? This would make the function more useful for other version numbers.

How about something like this?: `[rv]?(\\d+(.\\d+)*).*`

Reply:
Makes sense, thanks. Didn't not think about other versions that Icinga specific ones. This could work for plugin or output matches then too.

Reply:
`[rv]?(\d[.\d+]*).*` is the correct regex, as otherwise you would have a group extractor inside a group. This doesn't work well with the other proposal:

```
<1> => compare_version(AppVersion, "v2.8.1")
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error while evaluating expression: Can't convert '1-403' to an integer.
```